src/main/java/org/jboss/aerogear/keycloak/metrics/PrometheusExporter.java

```diff
@@ -49,18 +59,64 @@ public static PrometheusExporter instance() {
     return INSTANCE;
   }
 
+  // Creates a counter based on a event name
+  private static Counter createCounter(String name, boolean isAdmin) {
+    Counter.Builder counter = Counter.build().name(name);
+
+    if (isAdmin) {
+      counter.labelNames("realm", "resource").help("Generic KeyCloak Admin event");
+    } else {
+      counter.labelNames("realm").help("Generic KeyCloak User event");
+    }
+
+    return counter.register();
+  }
+
+  /**
+   * Count generic user event
+   *
+   * @param event User event
+   */
+  public void recordUserEvent(Event event) {
+    String eventName = USER_EVENT_PREFIX + event.getType().name();
+    counters.get(eventName).labels(event.getRealmId()).inc();
+  }
+
+  /**
+   * Count generic admin event
+   *
+   * @param event Admin event
+   */
+  public void recordAdminEvent(AdminEvent event) {
+    String eventName = ADMIN_EVENT_PREFIX + event.getOperationType().name();
```

Comment:
final ... as said in the other comment - I personally like to make the final more obvious